src/communication/index.ts

```diff
@@ -1,108 +1,75 @@
-import {
-  InterfaceMessageIds,
-  InterfaceMessageEvent,
-  SentSDKMessage,
-  SDKMessageIds,
-  SDKMessageToPayload,
-  RequestId,
-  InterfaceMessageToPayload,
-  Communicator,
-} from '../types';
-import { INTERFACE_MESSAGES } from './messageIds';
-
-class InterfaceCommunicator implements Communicator {
+import semver from 'semver';
+import { InterfaceMessageEvent, Communicator, Methods, SDKRequestData } from '../types';
+import { generateRequestId, DEFAULT_ALLOWED_ORIGINS } from './utils';
+
+// eslint-disable-next-line
+type Callback = (response: any) => void;
+
+class PostMessageCommunicator implements Communicator {
   private allowedOrigins: RegExp[] = [];
+  private callbacks = new Map<string, Callback>();
+  sdkVersion: string;
 
-  constructor(allowedOrigins: RegExp[]) {
-    this.allowedOrigins = allowedOrigins;
+  constructor(allowedOrigins: RegExp[], sdkVersion: string) {
+    this.allowedOrigins = [...DEFAULT_ALLOWED_ORIGINS, ...allowedOrigins];
+    this.sdkVersion = sdkVersion;
 
     window.addEventListener('message', this.onParentMessage);
   }
 
-  private isValidMessage({ origin, data }: InterfaceMessageEvent): boolean {
-    const emptyOrMalformed = !data || !data.messageId;
+  private isValidMessage = ({ origin, data }: InterfaceMessageEvent): boolean => {
+    const emptyOrMalformed = !data;
     const unknownOrigin = this.allowedOrigins?.find((regExp) => regExp.test(origin)) === undefined;
```

Comment:
I would not use the origin, as we are limiting the usage of the sdk to our domains.

I would propose to check the event source against the top level window, so this way we make sure that the message is send by the root window: e.g. `window.top === event.source`